portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -33,32 +33,33 @@
 
     // Sorts the tasks in descending order based on duration.
     // Longest task comes first in the collection.
-    // Then sorts the list again by alphabetical order ascending.
+    // Then sorts the list again by task priority.
     Collections.sort(tasksList, sortByTaskDurationDescendingThenPriority);
 
     CalendarEventsGroup calendarEventsGroup =
         new CalendarEventsGroup(eventsList, workHoursStartTime, workHoursEndTime);
+
     // Create a TimeRangeGroup class for the free time ranges.
     List<TimeRange> availableTimes = calendarEventsGroup.getFreeTimeRanges();
     TimeRangeGroup availableTimesGroup = new ArrayListTimeRangeGroup(availableTimes);
 
     List<ScheduledTask> scheduledTasks = new ArrayList<ScheduledTask>();
 
     for (Task task : tasksList) {
-      Duration taskDuration = task.getDuration();
+
       Optional<TimeRange> timeRangeForTask = findTimeForTask(task, availableTimesGroup);
 
       if (!timeRangeForTask.isPresent()) {
-        continue;
+        // Try to break up the current task and schedule it as much as possible.
+        boolean isTaskCompletelyScheduled =
```

Comment:
Consider adding a TODO statement to handle isTaskCompletelyScheduled. I don't think it's used right now.

Reply:
Updated.